### PR TITLE
[Performance] xetla kernels benchmark integration

### PR DESCRIPTION
**Pull request by ESI-SYD**
This PR create the Triton benchmark python module for the first time. 

- Integrated the XeTLA kernels to the Triton benchmark.

We can run the benchmark as
```
import xetla_benchmark
import xetla_benchmark.fused_softmax

xetla_benchmark.fused_softmax.benchmark.run()
```

### CODE REVIEW

File: python/tutorials/install_xetla_kernels.sh [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+  echo "**** Default BASE is set to $BASE ****"
+fi
+
+XETLA_PROJ=$BASE/ipex-extension-example
+if [ ! -d "$XETLA_PROJ" ]; then
+  cd $BASE
+  git clone https://github.com/ESI-SYD/ipex-extension-example.git -b dev/triton-xetla
+fi
+cd $XETLA_PROJ
+git submodule update --init --recursive
```
Comment (pbchekin) [inline python/tutorials/install_xetla_kernels.sh:16]: This repository has a submodule in intel-innersource, this will not work without proper credentials.

Comment (chengjunlu) [reply python/tutorials/install_xetla_kernels.sh]: I removed the changes in the scripts to integrate the benchmark running in this PR. We need further discussion about how to add a new job in the CI for benchmark.

File: .gitmodules [hunk 2]
```diff
@@ -0,0 +1,3 @@
+[submodule "benchmark/third_party/xetla"]
```
Comment (pbchekin) [inline .gitmodules:1]: Please don't use submodules. Instead consider using an external repository in the same way as we currently use llvm, e2e benchmarks, and so on.

**Review by pbchekin — CHANGES_REQUESTED** [review]

File: .gitmodules (hunk 2 above)
Comment (chengjunlu) [reply .gitmodules]: I remove the git submodule. Download the XeTLA library when building.

**Comment by ESI-SYD** [comment]
Is there a runner that includes available oneapi `dnnl` which is a dependency of this benchmark?

```
/home/runner/intel/oneapi/dnnl/latest/env/vars.sh: No such file or directory
```

**Comment by chengjunlu** [comment]
Hi @pbchekin,
Could you please help to review the work flow yaml?

**Comment by pbchekin** [comment]
> Is there a runner that includes available oneapi `dnnl` which is a dependency of this benchmark?
> 
> ```
> /home/runner/intel/oneapi/dnnl/latest/env/vars.sh: No such file or directory
> ```

[PyTorch Prerequisites for Intel GPUs](https://www.intel.com/content/www/us/en/developer/articles/tool/pytorch-prerequisites-for-intel-gpus.html) does not have dnnl, I think, but you can install one in the runner in runtime (with sudo apt update; sudo apt install ...), or we can have a custom runner image with pre-installed dnnl, see https://github.com/intel/intel-xpu-backend-for-triton/blob/llvm-target/.github/dockerfiles/runner-base/Dockerfile.

**Comment by ESI-SYD** [comment]
> > Is there a runner that includes available oneapi `dnnl` which is a dependency of this benchmark?
> > ```
> > /home/runner/intel/oneapi/dnnl/latest/env/vars.sh: No such file or directory
> > ```
> 
> [PyTorch Prerequisites for Intel GPUs](https://www.intel.com/content/www/us/en/developer/articles/tool/pytorch-prerequisites-for-intel-gpus.html) does not have dnnl, I think, but you can install one in the runner in runtime (with sudo apt update; sudo apt install ...), or we can have a custom runner image with pre-installed dnnl, see https://github.com/intel/intel-xpu-backend-for-triton/blob/llvm-target/.github/dockerfiles/runner-base/Dockerfile.

Thanks, workaround to use actually compatible 2024.1 oneapi basekit in runtime

File: python/tutorials/install_xetla_kernels.sh (hunk 1 above)
Comment (chengjunlu) [reply python/tutorials/install_xetla_kernels.sh]: The `triton-benchmark.yaml` has been added.

File: .github/workflows/triton-benchmarks.yml [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+      runner_label:
+        description: Runner label, keep empty for default
+        type: string
+        default: ""
+  pull_request:
+    branches:
+      - llvm-target
+  push:
+    branches:
+      - llvm-target
```
Comment (pbchekin) [inline .github/workflows/triton-benchmarks.yml:15]: Do you want to remove this before merging? Running manually is fine, you can also run it daily/nightly if you want but running for each commit is too much IMHO.

**Review by pbchekin — APPROVED** [review]
Looks good! You will probably need to disable running the workflow on each commit and add a cron similar to https://github.com/intel/intel-xpu-backend-for-triton/blob/llvm-target/.github/workflows/nightly-wheels.yml#L5-L6.

File: .github/workflows/triton-benchmarks.yml (hunk 3 above)
Comment (ESI-SYD) [reply .github/workflows/triton-benchmarks.yml]: Done

**Pull request merged**